### PR TITLE
github-ci:Create new Telegram Notifier action

### PR DESCRIPTION
**Pull request by drakonhg**
The problem was that previous action worked 50/50 or
did not work at all. So I decided to create a new action
which is functional and has the same process logic

Also two additional actions were added. They are required
for telegram action work better. First one is setting
additional environment variables that we are need for
message to be correctly structured and another is to check
if commit sha exist or not

All explanations can be found in README.md of each action

### CODE REVIEW

File: .github/actions/telegram-notifier/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Container image that runs your code
```
Comment (VitaliyaIoffe) [inline .github/actions/telegram-notifier/Dockerfile:1]: Why do we need Dockerfile?

File: .github/actions/telegram-notifier/script.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+MESSAGE = os.environ['INPUT_MESSAGE']
+PAGE_PREVIEW = os.environ['INPUT_DISABLE_WEB_PAGE_PREVIEW']
+BRANCH_NAME = os.environ['GITHUB_REF']
+ACTOR = os.environ['GITHUB_ACTOR']
+
+# Static request
+REQUEST = f'https://api.telegram.org/bot{TOKEN}/'
+
+
+def check_branch_name(branch_name):
```
Comment (VitaliyaIoffe) [inline .github/actions/telegram-notifier/script.py:17]: docstring, please

File: .github/actions/telegram-notifier/script.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        branch_name != 'refs/heads/1.10',
+        not branch_name.startswith('refs/heads/2.'),
+        not branch_name.startswith('refs/tags'),
+        ]
+
+    if any(branch_list):
+        return CHAT_ID + "_" + ACTOR
+    return CHAT_ID
+
+def send_message(chat_id, message,
```
Comment (VitaliyaIoffe) [inline .github/actions/telegram-notifier/script.py:29]: docstring, please

File: .github/actions/telegram-notifier/script.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+REQUEST = f'https://api.telegram.org/bot{TOKEN}/'
+
+
+def check_branch_name(branch_name):
+    branch_list = [
+        branch_name != 'refs/heads/master',
+        branch_name != 'refs/heads/1.10',
+        not branch_name.startswith('refs/heads/2.'),
+        not branch_name.startswith('refs/tags'),
+        ]
```
Comment (VitaliyaIoffe) [inline .github/actions/telegram-notifier/script.py:23]: You used different alignments in the file.
I think is more preferable: 
```
some = [
    foo, 
    bar
]
```

File: .github/actions/telegram-notifier/script.py [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+    if any(branch_list):
+        return CHAT_ID + "_" + ACTOR
+    return CHAT_ID
+
```
Comment (VitaliyaIoffe) [inline .github/actions/telegram-notifier/script.py:28]: `E302 expected 2 blank lines, found 1`

File: .github/actions/telegram-notifier/script.py [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+def send_message(chat_id, message,
+                 disable_web_page_preview=False, markdown='MarkdownV2'):
+    edit_message = message.replace('-', '\\-').replace('_', '\\_').replace('.', '\\.')
```
Comment (VitaliyaIoffe) [inline .github/actions/telegram-notifier/script.py:31]: Firstly, these replacements are not obvious.
Secondly, `E501 line too long (86 > 79 characters)`

File: .github/actions/telegram-notifier/script.py [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+    send_message_url = REQUEST + 'sendMessage'
+    send_message_data = {
+        'chat_id': chat_id,
+        'parse_mode': markdown,
+        'disable_web_page_preview': disable_web_page_preview,
+        'text': edit_message,
+    }
+
+    response = requests.post(send_message_url, send_message_data)
+    print(response.json())
```
Comment (VitaliyaIoffe) [inline .github/actions/telegram-notifier/script.py:42]: Do we need this print?

Comment (VitaliyaIoffe) [reply .github/actions/telegram-notifier/script.py:42]: Why have not we checked the response code?

File: .github/actions/telegram-notifier/action.yml [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+    required: true
+    description: 'Message that bot will send'
+  disable_web_page_preview:
+    required: false
+    description: 'Preview web page if False, else disable preview'
+
+# definition how we want our action
+runs:
+  using: docker # we are saying this action will run on docker
+  image: 'Dockerfile' # we will use our own Dockerfile to build an image and run it.
```
Comment (VitaliyaIoffe) [inline .github/actions/telegram-notifier/action.yml:23]: no newline at end of file

File: .github/actions/configure_github_envs/action.yml [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+      run: |
+        echo Running curl
+        sudo -n apt update -y ||:
+        sudo -n apt install -y jq ||:
+        echo 'JOB_ID<<EOF' >> $GITHUB_ENV
+        curl --header "Accept: application/vnd.github.v3+json" \
+        --header "Bearer: ${{ github.token }}" \
+        -s https://api.github.com/repos/drakonhg/telegram-notifier/actions/runs/${{ github.run_id }}/jobs | jq -r '.jobs[0].id' | tee -a $GITHUB_ENV
+        echo 'EOF' >> $GITHUB_ENV
+      shell: bash
```
Comment (VitaliyaIoffe) [inline .github/actions/configure_github_envs/action.yml:25]: `no newline at end of file`

File: .github/actions/commit-hash-check/action.yml [hunk 10]
```diff
@@ -0,0 +1,13 @@
...
+  commit:
+    description: 'Commit ID'
+    required: false
+runs:
+  using: 'composite'
+  steps:
+    - name: 'Check Commit SHA'
+      run: |
+        git checkout ${{ inputs.commit }}
+      shell: bash
```
Comment (VitaliyaIoffe) [inline .github/actions/commit-hash-check/action.yml:13]: `no newline at end of file`

**Review by VitaliyaIoffe — CHANGES_REQUESTED** [review]
Please, rebase on the master

**Comment by kyukhin** [comment]
I guess we should move to another messenger. Closing.

**Pull request closed without merging**